extensions/data-transfer/portability-data-transfer-google/src/main/java/org/datatransferproject/datatransfer/google/photos/GooglePhotosExporter.java

```diff
@@ -167,38 +174,125 @@ public GooglePhotosExporter(GoogleCredentialFactory credentialFactory, JsonFacto
+    GoogleMediaItem[] mediaItems = mediaItemSearchResponse.getMediaItems();
+    if (mediaItems != null && mediaItems.length > 0) {
+      List<PhotoModel> photos = convertPhotosList(albumId, mediaItems, jobId);
+      containerResource = new PhotosContainerResource(null, photos);
     }
 
-    PhotosContainerResource containerResource = new PhotosContainerResource(null, photos);
-
     ResultType resultType = ResultType.CONTINUE;
     if (nextPageData == null) {
       resultType = ResultType.END;
     }
+
     return new ExportResult<>(resultType, containerResource, continuationData);
   }
 
+
+  @VisibleForTesting
+  void populateContainedPhotosList(UUID jobId, TokensAndUrlAuthData authData)
+      throws IOException {
+    // Get list of all album ids
+    List<String> albumIds = new LinkedList<>();
+    AlbumListResponse albumListResponse = getOrCreatePhotosInterface(authData)
+        .listAlbums(Optional.empty());
+    albumIds.addAll(Arrays.stream(albumListResponse.getAlbums()).map(GoogleAlbum::getId)
+        .collect(Collectors.toList()));
+    String token;
+    while ((token = albumListResponse.getNextPageToken()) != null) {
+      albumListResponse = getOrCreatePhotosInterface(authData)
+          .listAlbums(Optional.of(token));
+      albumIds.addAll(
+          Arrays.stream(albumListResponse
+              .getAlbums())
+              .map(GoogleAlbum::getId)
+              .collect(Collectors.toList()));
+    }
+
+    // Get list of all ids belonging to photos belonging to albums
+    TempPhotosData tempPhotosData = jobStore
+        .findData(jobId, createCacheKey(), TempPhotosData.class);
+    if (tempPhotosData == null) {
+      tempPhotosData = new TempPhotosData(jobId);
+      jobStore.create(jobId, createCacheKey(), tempPhotosData);
+    }
+    for (String albumId : albumIds) {
+      MediaItemSearchResponse albumMediaItemResponse = getOrCreatePhotosInterface(authData)
+          .listMediaItems(Optional.of(albumId), Optional.empty());
+      if (albumMediaItemResponse.getMediaItems().length > 0) {
+        tempPhotosData.addAllContainedPhotoIds(
+            Arrays.stream(albumMediaItemResponse.getMediaItems())
+                .map(GoogleMediaItem::getId)
+                .collect(Collectors.toList()));
+      }
+      while ((token = albumMediaItemResponse.getNextPageToken()) != null) {
+        albumMediaItemResponse = getOrCreatePhotosInterface(authData)
+            .listMediaItems(Optional.of(albumId), Optional.of(token));
+        if (albumMediaItemResponse.getMediaItems().length > 0) {
+          tempPhotosData.addAllContainedPhotoIds(
+              Arrays.stream(albumMediaItemResponse.getMediaItems())
+                  .map(GoogleMediaItem::getId)
+                  .collect(Collectors.toList()));
+        }
+      }
+    }
+    jobStore.update(jobId, createCacheKey(), tempPhotosData);
+  }
+
+  private Optional<String> getPaginationToken(Optional<PaginationData> paginationData) {
```

Comment:
Maybe rename this to getPhotosPagnationToken?  right now I would think it would work with albums too.